code/game/objects/structures.dm

```diff
@@ -185,3 +188,72 @@
 	user.do_attack_animation(src)
 	spawn(1) qdel(src)
 	return 1
+
+/obj/structure/proc/can_visually_connect()
+	return anchored
+
+/obj/structure/proc/can_visually_connect_to(var/obj/structure/S)
+	return istype(S, src)
+
+/obj/structure/proc/update_connections(propagate = 0)
+	var/list/dirs = list()
+	var/list/other_dirs = list()
+
+	for(var/obj/structure/S in orange(src, 1))
+		if(can_visually_connect_to(S))
+			if(S.can_visually_connect())
+				if(propagate)
+					//S.update_connections() //Not here
+					S.update_icon()
+				dirs += get_dir(src, S)
+
+	if(!can_visually_connect())
+		connections = list("0", "0", "0", "0")
+		other_connections = list("0", "0", "0", "0")
+		return FALSE
+
+	for(var/direction in cardinal)
+		var/turf/T = get_step(src, direction)
+		var/success = 0
+		for(var/b_type in blend_objects)
+			if(istype(T, b_type))
+				success = 1
+				if(propagate)
+					var/turf/simulated/wall/W = T
+					if(istype(W))
+						W.update_connections(1)
+				if(success)
+					break

```

Comment:
That is why it breaks out of the for loop! Since the break has no effect on the if(). So yes that break breaks out of the floor loop, which is what it's `for` (pun definitely intended).

Reply:
Which is to say, the `if(success) break` immediately following that is unnecessary.

Reply:
Well, if it didn't have a break after it, it wouldn't break out of the outer loop, so it's necessary, since break just breaks the innermost loop. But I will rewrite it with labeled loops to be more clear.

Reply:
Actually the top one you're right, the top one (this one) was a copypaste of the bottom one probably and is unnecessary. The bottom one could be better written using labels.